packages/backend/src/workers/action.ts

```diff
@@ -29,33 +29,34 @@ type JobData = {
 export const worker = new Worker(
   'action',
   tracer.wrap('workers.action', async (job) => {
-    const {
-      stepId,
-      flowId,
-      executionId,
-      nextStep,
-      executionStep,
-      nextStepMetadata,
-    } = await processAction({ ...(job.data as JobData), jobId: job.id })
+    const jobData = job.data as JobData
 
-    if (executionStep.isFailed) {
-      await Execution.query().patch({ status: 'failure' }).findById(executionId)
-      return handleErrorAndThrow(executionStep.errorDetails)
-    }
-
-    const step = await Step.query().findById(stepId).throwIfNotFound()
+    const step = await Step.query().findById(jobData.stepId)
 
-    // dd-trace span tagging
     const span = tracer.scope().active()
     span?.addTags({
-      flowId,
-      executionId,
-      stepId,
-      appKey: step.appKey,
+      flowId: jobData.flowId,
+      executionId: jobData.executionId,
+      stepId: jobData.stepId,
+      actionKey: step?.key,
+      appKey: step?.appKey,
```

Comment:
nit: thinking if we can still keep `throwIfNotFound` to guarantee non-nullability of `step` (just in case for future code change)

```
const span = tracer.scope().active()
span?.addTags({
  flowId: jobData.flowId,
  executionId: jobData.executionId,
  stepId: jobData.stepId
})

const step = await Step.query().findById(jobData.stepId).throwIfNotFound()
span?.addTags({
  actionKey: step.key,
  appKey: step.appKey,
})

```

Reply:
.throwIfNotFound here will cause the retries attempts also. that's why i delegate the error throwing to `processAction` since it tries to query step there as well.

Reply:
fml. Good catch.

The crux of the issue is that with bullmq's approach, we retry _all_ errors by default, which is unintuitive for our use case. zzzzzzzz